newIDE/app/src/InstancesEditor/index.js

```diff
@@ -836,6 +877,41 @@ export default class InstancesEditor extends Component<Props> {
       return;
     }
 
+    if (
+      this.selectionRectangle.hasStartedSelectionRectangle() &&
+      this.selectionRectangle.selectionRectangleEnd
+    ) {
+      this.selectionRectangle.updateSelectionRectangle(
+        this.selectionRectangle.selectionRectangleEnd.x + deltaX,
+        this.selectionRectangle.selectionRectangleEnd.y + deltaY
+      );
+      return;
+    }
+
+    if (!hasCursorMovedSinceItIsDown) {
+      // Select the instance that will be moved.
+      this.props.instancesSelection.selectInstance({
+        instance,
+        multiSelect: this.keyboardShortcuts.shouldMultiSelect(),
+        layersLocks: this._getLayersLocks(),
+      });
+      if (this.props.onInstancesSelected) {
+        this.props.onInstancesSelected(
+          this.props.instancesSelection.getSelectedInstances()
+        );
+      }
+
+      if (this.keyboardShortcuts.shouldCloneInstances()) {
```

Comment:
Are you sure we can clone something if the cursor has NOT moved since it's down? Seems like weird as this will create invisible, surprising clones of objects?

Reply:
I see this was already the case before, but still asking.

Reply:
It's in the callback for mouse movement. I renamed the variable.